### PR TITLE
EOS-16248 User interface for cluster start CLI

### PR DESCRIPTION
**Pull request by shindeam**
## Problem Statement
<pre>
  <code>
  Story Ref (if any):
    EOS-16248 User interface for cluster start CLI
  </code>
</pre>
## Unit testing on RPM done
<pre>
  <code>
  No
  </code>
</pre>
## Problem Description
<pre>
  <code>
    Implement user interface for cluster start CLI
  </code>
</pre>
## Solution
<pre>
  <code>
    - Code implementation in cluster manager to point to the cluster controller
    - Code implementation in CortxClusterManager to process the request provided at CLI level
    - Set output to cli of our actual core library function response
  </code>
</pre>
## Unit Test Cases
<pre>
  <code>
    cd ~/cortx-ha
    ./jenkins/build.sh
    sudo yum erase cortx-ha*
    sudo yum install ~/cortx-ha/dist/rpmbuild/RPMS/x86_64/cortx-ha-2.0.0-9525df2.x86_64.rpm
    -bash-4.2$ sudo /opt/seagate/cortx/ha/bin/cortxha cluster start
    {"status": "InProgress", "msg": "Cluster start operation performed"}

   **Negative test case (command is not proper)**
   -bash-4.2$ sudo /opt/seagate/cortx/ha/bin/cortxha cluster sta
   usage: cortxha

    Use below command for more detail
    cortx --help cluster
       [-h] {add_node,remove_node,start,stop,status,shutdown} ...
cortxha

    Use below command for more detail
    cortx --help cluster: error: argument cluster_action: invalid choice: 'sta' (choose from 'add_node', 'remove_node', 'start', 'stop', 'status', 'shutdown')

    **Negative test case (Adding extra command line argument)**
    -bash-4.2$ sudo /opt/seagate/cortx/ha/bin/cortxha cluster start one
usage: cortxha

    Use below command for more detail
    cortx --help

     cortxha: error: unrecognized arguments: one
 
  </code>
</pre>

### CODE REVIEW

File: jenkins/pyinstaller/v2/pyinstaller-cortx-ha.spec [hunk 1]
```diff
@@ -21,16 +21,27 @@ import os
...
 ha_path="<HA_PATH>"
+product_module_list = import_list(ha_path, ha_path + "/ha/core")
 
 # Analysis
 cortxha =  Analysis([ha_path + '/ha/cli/cortxha.py'],
         pathex=[ha_path],
         binaries=[],
         datas=[],
-        hiddenimports=[],
+        hiddenimports=product_module_list,
```
Comment (azheregelya) [inline jenkins/pyinstaller/v2/pyinstaller-cortx-ha.spec:44]: What problem is solved by this move?

**Review by azheregelya — APPROVED** [review]

File: jenkins/pyinstaller/v2/pyinstaller-cortx-ha.spec (hunk 1 above)
Comment (shindeam) [reply jenkins/pyinstaller/v2/pyinstaller-cortx-ha.spec:44]: We were facing module not found issue in the dynamic import_module functionality hence in pyinstaller we need to add the hidden imports to catch those.

**Comment by ArchanaLimaye** [comment]
Please add details  of negative tests to Unit Test Cases section

File: ha/core/cluster/cluster_manager.py [hunk 2]
```diff
@@ -293,3 +294,24 @@ def controller_list(self) -> list:
...
+        Args:
+            action ([string]): Take cluster action for each request.
+            args ([dict]): Parameter pass to request to process.
+        """
+        self._output = output
+        # TODO Add validater
+        if action == const.CLUSTER_COMMAND:
+            if args.cluster_action == const.CLUSTER_ACTIONS.START.value:
+                Log.debug("Executing cortxha cluster start")
+                self._output.output(self._controllers[const.CLUSTER_CONTROLLER].start())
```
Comment (ArchanaLimaye) [inline ha/core/cluster/cluster_manager.py:311]: How is "pcs cluster start" executed from here ?

File: ha/core/cluster/cluster_manager.py [hunk 3]
```diff
@@ -293,3 +294,24 @@ def controller_list(self) -> list:
...
+        """
+        Generic method to handle process request
+
+        Args:
+            action ([string]): Take cluster action for each request.
+            args ([dict]): Parameter pass to request to process.
+        """
+        self._output = output
+        # TODO Add validater
+        if action == const.CLUSTER_COMMAND:
```
Comment (ArchanaLimaye) [inline ha/core/cluster/cluster_manager.py:308]: This code is temporary right ? can you please add comment stating the same here ?

Comment (shindeam) [reply ha/core/cluster/cluster_manager.py:309]: Done

**Comment by shindeam** [comment]
> Please add details of negative tests to Unit Test Cases section

Added negative tests cases.

**Review by indrajitzagade — APPROVED** [review]

**Comment by ajaysrivas** [comment]
@shindeam Cli framework PR has been merged. Please rebase with main and update the PR.

**Comment by shindeam** [comment]
Created another PR with updated code.

**Pull request closed without merging**